umm-spec-lib/src/cmr/umm_spec/migration/characteristics_data_type_migration.clj

```diff
@@ -0,0 +1,73 @@
+(ns cmr.umm-spec.migration.characteristics-data-type-migration
+  "Contains helper functions for migrating between different versions of UMM characteristics' data type."
+  (:require
+   [clojure.set :as set] 
+   [clojure.string :as string]
+   [cmr.common.util :as util :refer [update-in-each]]
+   [cmr.umm-spec.models.umm-common-models :as umm-cmn]
+   [cmr.umm-spec.util :as umm-spec-util]))
+
+(def mapping-up
+  "Defines mappings of Characteristics' data type values from v1.9 to v1.10.
+   Anything not in this map, and not equal to \"NOT APPLICABLE\", map it to \"STRING\"."
+  {"TIME/DIRECTION (ASCENDING)" "STRING"
+   "TIME/DIRECTION (DESCENDING)" "STRING"
+   "VARCHAR" "STRING"
+   "INTEGER" "INT"
+   "RADIOCARBON DATES" "STRING"
+   "STRING" "STRING"
+   "FLOAT" "FLOAT"
+   "INT" "INT"
+   "BOOLEAN" "BOOLEAN"
+   "DATE" "DATE"
+   "TIME" "TIME"
+   "DATETIME" "DATETIME"
+   "DATE_STRING" "DATE_STRING"
+   "TIME_STRING" "TIME_STRING"
+   "DATETIME_STRING" "DATETIME_STRING"})
+
+(defn migrate-data-type
+  "Migrate the Charateristics' data type from string to enum."
+  [characteristics]
+  (when-let [data-type (:DataType characteristics)]
```

Comment:
Since `DataType` is a required field, the `when` check is unnecessary.

Reply:
Good that you spotted that. I didn't think it's necessary. but I worked on part of the code, which I don't remember exactly where, it seems that it's possible that individual Characteristic could be nil?

Reply:
When `map` on nil, it does nothing. I don't think we should have a case of `[nil]` as the value of characteristics; If we do, we should fix the code that generated [nil], not here.

Reply:
Done.